### DISCUSSION TITLE
How to disable right click context menu or customize it in rust without change js code?

### DISCUSSION THREAD

**Discussion by crackleeeessyp** [tauri-apps/tauri#11808] [Q&A] [answered]
How to disable right click context menu or customize it in rust without change js code

**Comment by FabianLars** [comment]
you currently must use js to disable/overwrite the native context menu (listen to the `contextmenu` js event). you can however inject that from the rust side via https://docs.rs/tauri/2.0.0/tauri/webview/struct.WebviewWindowBuilder.html#method.initialization_script and https://docs.rs/tauri/2.0.0/tauri/webview/struct.WebviewWindow.html#method.eval

If you only care about windows, you can also use the webview rust bindings and do https://learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/context-menus?tabs=csharp#disabling-context-menus

**Reply by crackleeeessyp** [reply]
Sure, for the first solution, I'm a bit concerned it may pollute the front-end code.

I'm curious about the second solution. I'm not very familiar with Rust. Are the `webview Rust bindings` available in Tauri, and is there documentation on how to change it in a Tauri app? Or is it another framework similar to Tauri?

**Reply by FabianLars** [reply, marked as answer]
this is the entrypoint to use the bindings in tauri: https://docs.rs/tauri/2.0.0/tauri/webview/struct.WebviewWindow.html#method.with_webview
